akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -801,11 +801,25 @@ class PersistentShardCoordinator(typeName: String, settings: ClusterShardingSett
   }: Receive).orElse[Any, Unit](receiveTerminated).orElse[Any, Unit](receiveSnapshotResult)
 
   def receiveSnapshotResult: Receive = {
-    case SaveSnapshotSuccess(_) ⇒
+    case SaveSnapshotSuccess(m) ⇒
       log.debug("Persistent snapshot saved successfully")
+      deleteMessages(m.sequenceNr)
```

Comment:
reg. weaker consistency: if snapshots have weaker consistency than messages having a buffer doesn't help a lot and only creates an illusion of being safe. it can turn out that previous snapshot is not yet visible as well (or even the one before as well and so on). So as long guarantees for snapshots are weaker than for the messages for any N < Infinity there will be a situation, when that kind of mismatch will happen. in our devs we're treating snapshots and messages as same-class-citizens with same consistency to ensure safety.

for debugging purposes it's completely ok, but I propose to have that one configurable then with default value of 2 (batches).